src/openrct2/world/map_animation.c

```diff
@@ -456,7 +456,7 @@ static bool map_animation_invalidate_wall_door(sint32 x, sint32 y, sint32 baseZ)
 	if (gCurrentTicks & 1)
 		return false;
 
-	bool wasInvalidated = false;
+	bool removeAnimation = true;
```

Comment:
All of the animation functions either remove the animation entry at the end or keep it for the next frame to decide. Finish animation would not quite make sense in this situation.